### ISSUE TITLE
x/pkgsite: package search doesn't find main package with matching name

### ISSUE THREAD

**Issue by ComaVN** [golang/go#70459]
### What is the URL of the page with the issue?

https://pkg.go.dev/search?q=multee

### What is your user agent?

Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36

### Screenshot

![image](https://github.com/user-attachments/assets/3e3ae8c1-0328-4937-b8dd-8ca830a6755d)

### What did you do?

I published my package [multee](https://github.com/ComaVN/multee).

### What did you see happen?

pkg.go.dev doesn't show the [main package](https://github.com/ComaVN/multee/blob/v0.0.3/multee.go) when searching for "multee", but a sub-package (an alternative implementation that I kept around for benchmarking and testing)

### What did you expect to see?

I expect to see the search results show the main package, not a "random" sub package.

**Comment by ComaVN** [comment]
Some context: in the v0.0.1 version, the [sub-package with the alt implementation](https://github.com/ComaVN/multee/blob/v0.0.1/alt/byteslice/multee.go#L8) was also named `multee`, which I fixed in v0.0.3

If I search for the module in a different way, eg. ["multiplexer for io.Readers"](https://pkg.go.dev/search?q=multiplexer+for+io.Readers), it does show the main package.

**Comment by adonovan** [comment]
@jba

**Comment by findleyr** [comment]
Interestingly, a search for "ComaVN/multee" does return the package:
https://pkg.go.dev/search?q=ComaVN%2Fmultee&m=package

Threfore, this does look like a pkgsite bug (not just poor search ranking, which we have various other problems with).

**Comment by ComaVN** [comment]
If this cannot be easily fixed, would it be at least possible to remove my package, so I can add it again, without the confusingly named `/alt` packages? I think that would fix it as well, no?

**Comment by jba** [comment]
Here is why pkgsite shows the subpackage instead of the main one when you search for "multee":
1. Both have the same number of importers, so popularity doesn't affect the ranking.
2. The subpackage actually uses the word "multee" in its synopsis, while the main one doesn't (and both paths contain "multee" once).

So I think the search algorithm is working as intended. We deliberately don't have any rule that prefers "higher-up" or shorter import paths to longer ones, because there is no reason why the shorter path should be more popular than the longer one.

In other words, there is no way for pkgsite to know that what you call the main package is really the important one.

The best way to solve your problem is to get users. Presumably a lot more people will use the main package, and this increased popularity will push it to the top of the list.

**Comment by ComaVN** [comment]
FYI @jba I released v0.0.4 where I addressed (some of) your points, but there still seems to be an oddness to the search result: following the link above, <https://pkg.go.dev/search?q=multee>, still shows the "wrong" package @v0.0.3, but if I then switch to "Symbols" and back to "Packages" the correct package and version *is* shown.

(It changes the url to <https://pkg.go.dev/search?limit=25&m=package&q=multee>, so it's probably some cache that needs to time out. I tried using a different browser in in-cognito mode as well).

I'll check in a few days, and if it doesn't go away, I'll request a package removal, and re-publish it.

**Comment by jba** [comment]
I think it was caching. I just tried the search and I got the v0.0.4 main package.

**Comment by ComaVN** [comment]
Same, thanks for your trouble
